lib/archethic/contracts/worker.ex

```diff
@@ -103,11 +103,16 @@ defmodule Archethic.Contracts.Worker do
       contract_transaction = Constants.to_transaction(contract_constants)
 
       with true <- ConditionInterpreter.valid_conditions?(transaction_condition, constants),
-           next_tx <- ActionInterpreter.execute(Map.fetch!(triggers, :transaction), constants),
+           next_tx when not is_nil(next_tx) <-
```

Comment:
It might be better to pattern match on the Transaction struct instead of `not nil`